hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncConnectionImpl.java

```diff
@@ -326,15 +328,17 @@ CompletableFuture<MasterService.Interface> getMasterStub() {
           }
         }
 
-      });
+      }, conf.getInt(HConstants.CONNECTION_REGISTRY_API_TIMEOUT,
```

Comment:
I do not think this is the correct place to put these timeout configurations.

We should place them into the connection registry implementation.